lib/get-config.js

```diff
@@ -133,6 +125,12 @@ function validateRootProperties(source) {
     if (!KNOWN_ROOT_PROPERTIES.has(key)) {
       throw new Error(`Unknown top-level configuration property detected: ${key}`);
     }
+
+    if ('pending' in source) {
+      throw new Error(
```

Comment:
This new exception is dead code because the above exception will fire first when it encounters the unknown `pending` property. Personally, I think the existing exception is sufficient, and we don't necessarily need to add a new one.

Reply:
Hmmm good point. I can remove it.